src/api/api.c

```diff
@@ -788,6 +788,7 @@ z_get_options_t z_get_options_default(void) {
 #if Z_FEATURE_ATTACHMENT == 1
         // TODO:ATT.attachment = z_attachment_null()
 #endif
+        .timeout_ms = 0
```

Comment:
To make it consistent with Zenoh it should have the same default value of 10 seconds. Although you might need to double check that value.